### DISCUSSION TITLE
Get SubmitType value

### DISCUSSION THREAD

**Discussion by homersimpsons** [symfony/symfony#59155] [Q&A]
In order to reset the pagination during "search" button click I want to add a SubmitType value.

I created a Form model:
```php
class UserSearch
{
    public string|null $email = null;
    public int $offset = 0;
}
```

And a FormType:
```php
class UserSearchType extends AbstractType
{
    /** @inheritDoc */
    public function buildForm(FormBuilderInterface $builder, array $options): void
    {
        $builder
            ->add('email', EmailType::class, ['required' => false])
            ->add('offset', SubmitType::class, ['label' => 'Rechercher', 'attr' => ['value' => 0]]);
    }

    public function configureOptions(OptionsResolver $resolver): void
    {
        $resolver->setDefaults([
            'data_class' => UserSearch::class,
            'method' => 'GET',
            'csrf_protection' => false, // Remove the CSRF token in query parameters
        ]);
    }

    public function getBlockPrefix(): string
    {
        return ''; // Used to avoid `user_search[email]=...` in query parameters
    }
}
```

The `offset=0` query parameter is correctly passed, but it is not hydrated in my `UserSearch` model, because when I use `offset=10`, my `$userSearch->offset` is still `0`.

Do you know if this is expected? Do you have a workaround?

I cannot go with an hidden field as its value will be updated on page change, I have to force it to `0`.

**Comment by MatTheCat** [comment]
In Symfony buttons cannot hold data: https://github.com/symfony/symfony/blob/2d5478644756f3207f4b3906d197c7721abf857b/src/Symfony/Component/Form/Button.php#L160-L164

I suggest you use the `MapQueryString` attribute instead so that you can handle the search through a form and the page through links.

**Reply by homersimpsons** [reply]
But with a `MapQueryString` I do not benefit from symfony's form rendering and validation directly.

**Reply by MatTheCat** [reply]
MapQueryString will validate your DTO.

As for the rendering, I never saw pagination embedded in a form. Navigation is usually implemented using links.

**Reply by homersimpsons** [reply]
> As for the rendering, I never saw pagination embedded in a form. Navigation is usually implemented using links.

Yes, I implement this as links for the pagination component. But on my page I have a search component with inputs (name, date, amount ...) so I just want to reset the navigation to  `0` when someone submit this form AND to not bother with the rendering of it so it is rendered like any other form.

**Comment by stof** [comment]
Buttons are handled as unmapped fields. I suggest you to use a hidden input for your `offset` instead.

**Reply by homersimpsons** [reply]
Do you know how could I reset this field / force its value to alwas be 0?
